### DISCUSSION TITLE
GPU computation on pi?

### DISCUSSION THREAD

**Discussion by storxusmc** [geerlingguy/raspberry-pi-pcie-devices#212] [General]
Hey All, 

I know this may be a controversial topic to discuss, but I was curious if this may be possible. I have a crypto-mining operation that I invested in back in 2015 and have been running multiple GPUs in a solo mining operation using solar power. 

I have been running these rigs off 2015/16 Celeron setups for all these years, but I have had a few motherboard failures this past year and the ability to get computer hardware, in general, is difficult, but most don't even stock parts for these older setups.. so I have been limping along buying used hardware.

I run all the rigs on awesome miner using the Linux clients all controlled by my windows 10 VM. 

I know getting graphic acceleration has been a difficult task with the PCI slot on the raspberry compute i/o boards, but I was curious if potentially possible to purely use it for computing. I was thinking this over and just going off how little cpu usage and ram are needed to actually process the computations to the GPU and back to the internet. It would be cool if I could rebuild my mining operation with raspberry pi modules controlling GPUs for mining. 

The savings alone would be a huge benefit, because not only would it save me on the cost of motherboards and such trying to rebuild my rigs, the electric reduction alone would be a huge saving. 

Bonus: It would be cool if I could do all of this involving Kubernetes also.. and maybe dually use the pi modules for other tasks.

**Comment by geerlingguy** [comment]
The drivers all seem to balk at certain parts of the initialization process, on every GPU I've tested so far, so even if you ignore the graphics portions entirely, none of the cards have been able to do anything else either (I tested the CUDA cores on Nvidia and some other software too on the AMD cards and everything resulted in the same total system lockup.

**Reply by storxusmc** [reply]
Hmm, maybe running a windows os would make a difference. I really don't need the system to work hard once its booted into the OS and controlling the data to the GPUs because for example viewing the CPU usage on a 6 x GPU Celeron rig I have which only has 2 cores, it sits around 5% cpu usage as average.. so not much is performed on the cpu level outside of routing traffic to and from the gpu and to the internet.
